src/fs/data_cluster_contents_info.hh

```diff
@@ -52,25 +51,18 @@ public:
         return _up_to_date_data_size == 0;
     }
 
-    void make_read_only() noexcept {
-        _read_only = true;
-    }
-
-    bool can_compact() const noexcept {
-        return _read_only;
-    }
-
     void add_data(disk_offset_t disk_offset, inode_t inode, file_range data_range) noexcept {
-        {
-            auto it =_data.upper_bound(disk_offset);
-            assert(it == _data.end() || it->first >= disk_offset + data_range.size());
-            if (it != _data.begin()) {
-                it--;
-                assert(disk_offset >= it->first + it->second.data_range.size());
-            }
-        }
-        _data.emplace(disk_offset, cluster_data_vec {inode, data_range});
+        auto [it, succ] = _data.emplace(disk_offset, cluster_data_vec {inode, data_range});
+        assert(succ);
         _up_to_date_data_size += data_range.size();
+        if (it != _data.begin()) {
+            it--;
```

Comment:
@psarna what is your opinion about using pre-increment instead of post-increment for iterator types? This may be relevant in the debug build where the compiler does not optimize it away.